pkg/kubelet/cm/pod_container_manager_linux_test.go

```diff
@@ -293,3 +293,86 @@ func TestGetPodContainerName(t *testing.T) {
 		})
 	}
 }
+
+func Test_podContainerManagerImpl_EnsureExists(t *testing.T) {
+	newGuaranteedPodWithUID := func(uid types.UID) *v1.Pod {
+		return &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID: uid,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name: "container",
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceCPU:    resource.MustParse("1000m"),
+								v1.ResourceMemory: resource.MustParse("1G"),
+							},
+							Limits: v1.ResourceList{
+								v1.ResourceCPU:    resource.MustParse("1000m"),
+								v1.ResourceMemory: resource.MustParse("1G"),
+							},
+						},
+					},
+				},
+			},
+		}
+	}
+
+	type fields struct {
+		cgroupManager CgroupManager
+	}
+	type args struct {
+		pod *v1.Pod
+	}
+	tests := []struct {
+		name            string
+		fields          fields
+		args            args
+		wantErr         bool
+		wantCgroupCalls []string
+	}{
+		{
+			name: "cgroup exists",
+			fields: fields{
+				cgroupManager: &FakeCgroupManager{exists: true},
+			},
+			args:            args{pod: newGuaranteedPodWithUID("fake-uid-1")},
+			wantErr:         false,
+			wantCgroupCalls: []string{"Exists"},
+		},
+		{
+			name: "cgroup created",
+			fields: fields{
+				cgroupManager: &FakeCgroupManager{exists: false},
+			},
+			args:            args{pod: newGuaranteedPodWithUID("fake-uid-1")},
+			wantErr:         false,
+			wantCgroupCalls: []string{"Create"},
+		},
+		{
+			name: "failed to create cgroup",
+			fields: fields{
+				cgroupManager: &FakeCgroupManager{exists: false, create: errors.New("test - failed to create cgroup")},
+			},
+			args:            args{pod: newGuaranteedPodWithUID("fake-uid-1")},
+			wantErr:         true,
+			wantCgroupCalls: []string{"Create"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			m := &podContainerManagerImpl{
+				cgroupManager: tt.fields.cgroupManager,
+			}
+			if err := m.EnsureExists(tt.args.pod); (err != nil) != tt.wantErr {
+				t.Errorf("podContainerManagerImpl.EnsureExists() error = %v, wantErr %v", err, tt.wantErr)
+			}
```

Comment:
nit:
```suggestion
			err := m.EnsureExists(tt.args.pod)
			assert.Equalf(t, tt.wantErr, err != nil, "podContainerManagerImpl.EnsureExists() error = %v, wantErr %v", err, tt.wantErr)
```

Reply:
Fixed in https://github.com/kubernetes/kubernetes/pull/127819/commits/35654a78549f6d26d460d58773db0f01c2f075bd